src/aws/cloudwatch-sink.cpp

```diff
@@ -222,6 +222,7 @@ void CloudWatchSink::cloudwatch_logs_thread()
             report_logger_error("Encountered an error while sending log events", "", e.what());
         }
     }
+    std::unique_lock<std::mutex> lock(logs_mtx_.get());
```

Comment:
Why not limit the scope of the lock just to the `try` block, same as above?

Reply:
dont wanna re-lock every loop operation since its only cleanup, added try except on the entire block